File: test/cpp/test_49.cpp
#include "../../src/cpp/code_49.cpp"
#include "cpp_deps/boilerplate.h"

void test(Solution& sol, const json& input, const json& output) {
    vector<string> strs = input["strs"].get<vector<string>>();
    vector<vector<string>> expected = output.get<vector<vector<string>>>();
    vector<vector<string>> result = sol.groupAnagrams(strs);

    for (auto& v : result) sort(v.begin(), v.end());

    sort(result.begin(), result.end(), [](const vector<string>& a, const vector<string>& b) {
        if (a.size() < b.size()) return true;
        else if (b.size() < a.size()) return false;
        else return a < b;
    });

    CHECK_EQ(result, expected);
}

TEST_CASE("") {
    TEST("test/test_json/test_49.json");
}

File: test/python/test_49.py
import sys
import os

sys.path.append(os.getcwd())
from .python_deps.boilerplate import BoilerTest
from src.python.code_49 import Solution

from typing import List


def input_fetcher(case):
    return case["input"]["strs"]


def nested_list_contents_comparator(
    answer: List[List[str]], actual: List[List[str]]
) -> bool:
    if len(answer) != len(actual):
        return False

    answer.sort()
    actual.sort()

    for ans, act in zip(answer, actual):
        if ans.sort() != act.sort():
            return False

    return True


def test():
    test = BoilerTest(
        49, input_fetcher, output_comparator=nested_list_contents_comparator
    )
    solution = Solution().groupAnagrams
    test.run_tests(solution)
